app/javascript/spec/namespace-form/namespace-form.spec.js

```diff
@@ -0,0 +1,148 @@
+        component: 'switch-field',
+        label: 'Enabled:',
+        name: 'enabled',
+      })];
+
+    const { fields } = schema(true, false);
+    expect(fields).toEqual(expectedResult);
+  });
+
+  it('should correctly initialize the component for namespace', (done) => {
+    const initialProps = {
+      isDomain: false,
+      namespaceId: 1111,
+      parentId: 2222,
+    };
+
+    fetchMock
+      .getOnce('/api/automate_namespaces/1111?attributes=name,description,enabled,source',
+        {
+          name: 'f', description: 'a', enabled: null, source: 'user',
+        });
+
+    const wrapper = mount(<NamespaceForm {...initialProps} />);
+    setImmediate(() => {
+      wrapper.update();
+      expect(wrapper.find('label').first().text()).toEqual('Name:');
+      expect(wrapper.find('input').first().instance().value).toEqual('f');
+      done();
+    });
+  });
+
+  it('should correctly initialize the component for domain', (done) => {
+    fetchMock
+      .getOnce('/api/automate_domains/2222?attributes=name,description,enabled,source',
+        {
+          name: 's', description: 'z', enabled: true, source: 'user',
+        });
+
+    const wrapper = mount(<NamespaceForm {...defaultProps} />);
+    setImmediate(() => {
+      wrapper.update();
+      expect(wrapper.find('input').first().props().value).toEqual('s');
+      expect(wrapper.find('input').last().props().value).toBeTruthy();
+      done();
+    });
+  });
+
+  it('should send correct data on submit', (done) => {
+    const initialProps = {
+      isDomain: false,
+      namespaceId: null,
+      parentId: 2222,
+    };
+    fetchMock.getOnce('/api/automate_domains/2222?attributes=name,description,enabled,source',
+      {
+        name: 's', description: 'z', enabled: true, source: 'user',
+      }).post('/api/automate_namespaces/', {});
+
+    const wrapper = mount(<NamespaceForm {...initialProps} />);
+    setImmediate(() => {
+      const Form = wrapper.find(FormRenderer).childAt(0);
+      Form.instance().form.change('name', 'New');
+      Form.instance().form.change('description', 'This is really New');
+      wrapper.update();
+      wrapper.find('button').at(0).simulate('click');
+      setImmediate(() => {
+        // I have no idea how to test it. The Object.defineProperty is not working.
+        // expect(window.location.href).toHaveBeenCalledWith('/miq_ae_class/explorer');
+        done();
+      });
+    });
+  });
+
+  it('should reset the form values and add a flash message', (done) => {
+    fetchMock
+      .getOnce('/api/automate_domains/2222?attributes=name,description,enabled,source',
+        {
+          name: 's', description: 'z', enabled: true, source: 'user',
+        });
+
+    const wrapper = mount(<NamespaceForm {...defaultProps} />);
+    setImmediate(() => {
+      wrapper.update();
+      const Form = wrapper.find(FormRenderer).childAt(0);
+      expect(Form.instance().form.getState().values.name).toEqual('s');
```

Comment:
You should not be checking the form state but the actual value of the input/Component in DOM.